packages/assets-controllers/src/assetsUtil.ts

```diff
@@ -292,3 +296,49 @@ export async function reduceInBatchesSerially<
   const finalResult = workingResult as Result;
   return finalResult;
 }
+
+/**
+ * fetches and maps the exchange rates
+ * @param nativeCurrency - native currency
+ * @param tokenAddresses - array of token addresses
+ * @param chainId - The ID of the chain.
+ */
+export async function fetchAndMapExchangeRates(
+  nativeCurrency: string,
+  tokenAddresses: Hex[],
+  chainId: Hex,
+): Promise<ContractExchangeRates> {
+  const tokenPricesFetchService: AbstractTokenPricesService =
+    codefiTokenPricesServiceV2;
+  // check if chainId is supported or currency is supported
+  const isChainIdSupported =
+    tokenPricesFetchService.validateChainIdSupported(chainId);
+  const isCurrencySupported =
+    tokenPricesFetchService.validateCurrencySupported(nativeCurrency);
+
+  if (!isChainIdSupported || !isCurrencySupported) {
+    return tokenAddresses.reduce((obj, tokenContractAddress) => {
+      return {
+        ...obj,
+        [tokenContractAddress]: undefined,
+      };
+    }, {});
```

Comment:
I see this was also taken from TokenRatesController, but is there value in setting these values to `undefined`? That essentially means the same thing as the properties not being there, so would it break anything to return an empty object instead? That would make this code clearer.

``` suggestion
      return {};
```